src/Nethermind/Nethermind.TxPool/Filters/DeployedCodeFilter.cs

```diff
@@ -3,23 +3,22 @@
 
 using Nethermind.Core;
 using Nethermind.Core.Specs;
+using Nethermind.Evm;
+using Nethermind.State;
 
 namespace Nethermind.TxPool.Filters
 {
     /// <summary>
     /// Filters out transactions that sender has any code deployed. If <see cref="IReleaseSpec.IsEip3607Enabled"/> is enabled.
     /// </summary>
-    internal sealed class DeployedCodeFilter : IIncomingTxFilter
+    internal sealed class DeployedCodeFilter(IWorldState worldState, ICodeInfoRepository codeInfoRepository, IChainHeadSpecProvider specProvider) : IIncomingTxFilter
     {
-        private readonly IChainHeadSpecProvider _specProvider;
-
-        public DeployedCodeFilter(IChainHeadSpecProvider specProvider)
-        {
-            _specProvider = specProvider;
-        }
         public AcceptTxResult Accept(Transaction tx, ref TxFilteringState state, TxHandlingOptions txHandlingOptions)
         {
-            return _specProvider.GetCurrentHeadSpec().IsEip3607Enabled && state.SenderAccount.HasCode
+            return specProvider.GetCurrentHeadSpec().IsEip3607Enabled
```

Comment:
Worth reusing IsValidTxSender?

Reply:
changed to reuse check from `IReadOnlyStateProviderExtensions`